### ISSUE TITLE
DatePicker: week count, date&time picker focus, quarter range & placeholder

### ISSUE THREAD

**Issue by anddrzejb** [ant-design-blazor/ant-design-blazor#1429]
### Describe the bug
I found several bugs in date/range picker.
~1. Week picker - week count for beginning of the year is showing incorrectly. For example first week in 2020 is actually starting from the second~
![image](https://user-images.githubusercontent.com/6518006/116353249-183bbf80-a7ff-11eb-87cc-e305e10982cc.png)
~and it should show like this:~
![image](https://user-images.githubusercontent.com/6518006/116353437-6d77d100-a7ff-11eb-91a9-2ae8107ab552.png)
Solved by #1452 

2. Date & Time picker - time is not focused and when changing time using keyboard, it does not focus either:
![image](https://user-images.githubusercontent.com/6518006/116353568-aca62200-a7ff-11eb-8bf0-aaac1618f12e.png)

3. Date & Time range picker - after selecting the date and clicking `Ok` button, the focus should move to the second date. Instead, the picker is closed and focus is taken off the input:
![dateTimePickerRange_2ndPickerFocus](https://user-images.githubusercontent.com/6518006/116354069-892fa700-a800-11eb-9ee5-0f2f3cf28763.gif)
Expected behaviour:
![dateTimePickerRange_2ndPickerFocus_Expected](https://user-images.githubusercontent.com/6518006/116354149-a8c6cf80-a800-11eb-9b2b-4298bace731a.gif)

4. Quarter range: When I set the end quarter, the same quarter of the start quarter is disabled, it should be enabled.
![image](https://user-images.githubusercontent.com/10374056/116251711-818ce580-a7a1-11eb-9e3d-20774be17a40.png)

5. Placeholder issue. 
  - the attribute `Placeholder` has no effect unless it is placed after `PickerType`
  - if it is visible, it is only visible after initialization of the component. If the value is set and the cleared, default placeholder is shown
![datePickerRange_vanishingPlaceholder](https://user-images.githubusercontent.com/6518006/116356879-81720180-a804-11eb-9f9c-9d2cdfca978e.gif)

### Steps to reproduce (please include code)
Except for the 5th point, all the other are reproduceable using the docs.
To reproduce the 5th, use this code (last 2 pickers will have a custom placeholder and it will vanish if cleared):
```razor
<DatePicker TValue="DateTime?" Placeholder="@("Will vanish on clear")" AllowClear Picker="@DatePickerType.Date" />
<DatePicker TValue="DateTime?" AllowClear Placeholder="@("Will vanish on clear")" />
<br/><br/>
<DatePicker TValue="DateTime?" Picker="@DatePickerType.Date" Placeholder="@("Will vanish on clear")" AllowClear/>
<DatePicker TValue="DateTime?" Picker="@DatePickerType.Date" AllowClear Placeholder="@("Will vanish on clear")"/>
```
### Further technical details
- AntDesign Nuget Package version: 0.8

**Comment by Zonciu** [comment]
1. When first clicked, the RangePicker panel will first display on the left side of the page.

2. The height of the RangePicker is refreshed twice when it is opened.

https://user-images.githubusercontent.com/10374056/116359573-88fbcf80-a831-11eb-8166-685ace5b25f4.mp4

**Comment by anddrzejb** [comment]
Just noticed one more - when no date is selected yet, the picker already shows current date as selected:
![image](https://user-images.githubusercontent.com/6518006/116361903-2c38ee80-a80a-11eb-9e13-bd38a69ed1bd.png)

It should look like this:
![image](https://user-images.githubusercontent.com/6518006/116361968-3b1fa100-a80a-11eb-8e89-52d1ff251ca3.png)

This issue applies to all type of pickers.

**Comment by anddrzejb** [comment]
Quoted from [this PR #1431](https://github.com/ant-design-blazor/ant-design-blazor/pull/1452#issuecomment-833405696)
> By the way, I found a problem, but this problem exists in the current version(0.8.0).
> RangePicker(date mode with timer) cannot select the correct end date, nor can you select months forward.
> Steps:
> 
> 1. Choose a start date
> 2. Select an end date, and you will find that the end date is always one month longer than the selected date, and you cannot switch to the previous month
> 
> -----------translate-----------
> 顺便提一下，我发现一个问题，但这个问题在当前版本(0.8.0)中就存在了。
> RangePicker（带时间的日期模式）不能选择正确的结束日期，也无法向前选择月份。
> 
> 步骤：
> 
> 1. 选择一个开始日期
> 2. 选择一个结束日期，此时会发现，结束日期会永远比选择的日期多1个月，并且无法切换到上一个月份
> 
> ![issue](https://user-images.githubusercontent.com/3509249/117279975-b9c4b000-ae94-11eb-9bbc-e0d618e9d4b2.gif)

**Comment by ElderJames** [comment]
Hi @Alexbits , this issue also addresses some issues around datepicker. I think some of them have already been solved. Could you please take a look at it when you're free?

**Comment by Alexbits** [comment]
> 4. Quarter range: When I set the end quarter, the same quarter of the start quarter is disabled, it should be enabled.
>    ![image](https://user-images.githubusercontent.com/10374056/116251711-818ce580-a7a1-11eb-9e3d-20774be17a40.png)
> 5. Placeholder issue.
> 
> * the attribute `Placeholder` has no effect unless it is placed after `PickerType`
> * if it is visible, it is only visible after initialization of the component. If the value is set and the cleared, default placeholder is shown
>   ![datePickerRange_vanishingPlaceholder](https://user-images.githubusercontent.com/6518006/116356879-81720180-a804-11eb-9f9c-9d2cdfca978e.gif)
> ### Steps to reproduce (please include code)
> Except for the 5th point, all the other are reproduceable using the docs. To reproduce the 5th, use this code (last 2 pickers will have a custom placeholder and it will vanish if cleared):
> 
> ```
> <DatePicker TValue="DateTime?" Placeholder="@("Will vanish on clear")" AllowClear Picker="@DatePickerType.Date" />
> <DatePicker TValue="DateTime?" AllowClear Placeholder="@("Will vanish on clear")" />
> <br/><br/>
> <DatePicker TValue="DateTime?" Picker="@DatePickerType.Date" Placeholder="@("Will vanish on clear")" AllowClear/>
> <DatePicker TValue="DateTime?" Picker="@DatePickerType.Date" AllowClear Placeholder="@("Will vanish on clear")"/>
> ```
> 
> ### Further technical details
> * AntDesign Nuget Package version: 0.8

@ElderJames These two issues are still present. I will look into this.

**Comment by Alexbits** [comment]
Hey @ElderJames. The `4. Quarter range issue` fixed in #2570. Please check.

**Comment by ElderJames** [comment]
@Alexbits Thank you!
